app/src/main/java/at/tugraz/ist/guessingwords/data/database/GWDatabase.kt

```diff
@@ -18,20 +18,38 @@ abstract class GWDatabase  : RoomDatabase() {
 
     companion object {
         var _instance: GWDatabase? = null
+        var _in_memory_instance: GWDatabase? = null
 
         fun getInstance(context: Context): GWDatabase {
-            if (_instance == null) {
+            if (_in_memory_instance == null){
+                if (_instance == null) {
```

Comment:
I have the feeling that this will cause terrible bugs later because the method does not do what you expect it to do.

I think it would be better, if the `getInstance` method always returns the persistent db.
Instead, in the WordService class, write the code into `getAllWords`, that first checks if the inMemory DB is there and uses it if it is, otherwise uses the persistent db.

Because otherwise, someone might call insertIntoDb and actually insert into the inMemoryDb instead of the persistent one...

If you change this back, you probably also have to review your tests to make sure they work in this new context.